src/syrupy/assertion.py

```diff
@@ -264,6 +264,7 @@ def __call__(
         extension_class: Optional[Type["AbstractSyrupyExtension"]] = None,
         matcher: Optional["PropertyMatcher"] = None,
         name: Optional["SnapshotIndex"] = None,
+        **kwargs: Any,
```

Comment:
I'm a bit worried about name collisions by just forwarding arbitrary arguments, vs. something like:

```
assert "hi" == snapshot(extra={...})
```

I know it doesn't read as nicely, but you can always add syntactic sugar via a fixture:

```
@pytest.fixture
def snapshot(named_arg: str):
    return snapshot.with_defaults(extra={ "named_arg": named_arg })

def test_case(snapshot):
    assert "hi" == snapshot(named_arg="hi")
```

Reply:
I added the argument to both `__call__` and `with_defaults` method. I added it to the `__call__` argument because I wanted it to work with cases when I want to change the snapshot behaviour but don't want it to persist.